beacon-chain/endtoend/endtoend_test.go

```diff
@@ -0,0 +1,503 @@
+package endtoend
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math/big"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/bazelbuild/rules_go/go/tools/bazel"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/pborman/uuid"
+	contracts "github.com/prysmaticlabs/prysm/contracts/deposit-contract"
+	"github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"github.com/prysmaticlabs/prysm/shared/testutil"
+	"google.golang.org/grpc"
+)
+
+var eth1BlockTime = uint64(1)
+
+type end2EndConfig struct {
+	minimalConfig  bool
+	tmpPath        string
+	epochsToRun    uint64
+	numValidators  uint64
+	numBeaconNodes uint64
+	contractAddr   common.Address
+	evaluators     []evaluator
+}
+
+type beaconNodeInfo struct {
+	processID   int
+	datadir     string
+	rpcPort     uint64
+	monitorPort uint64
+	grpcPort    uint64
+	multiAddr   string
+}
+
+func TestEndToEnd_DemoConfig(t *testing.T) {
+	tmpPath := path.Join("/tmp/e2e/", uuid.NewUUID().String()[:18])
+	os.MkdirAll(tmpPath, os.ModePerm)
+	fmt.Printf("Test Path: %s\n", tmpPath)
+
+	demoConfig := &end2EndConfig{
+		tmpPath:        tmpPath,
+		minimalConfig:  false,
+		epochsToRun:    8,
+		numBeaconNodes: 2,
+		numValidators:  8,
+		evaluators: []evaluator{
+			{
+				name:       "activate_validators",
+				policy:     onGenesisEpoch,
+				evaluation: validatorsActivate,
+			},
+			{
+				name:       "finalize_checkpoint",
+				policy:     afterNEpochs(4),
+				evaluation: finalizationOccurs,
+			},
+			// Evaluator{
+			//	Name:       "validators_participate",
+			// 	Policy:     AfterNEpochs(4),
+			// 	Evaluation: ValidatorsParticipating,
+			// },
+		},
+	}
+	runEndToEndTest(t, demoConfig)
+}
+
+func TestEndToEnd_MinimalConfig(t *testing.T) {
```

Comment:
With these e2e tests, it's probably a good idea to put them into individual files as well so they can be sharded and cached. 
```
demo_e2e_test.go
minimal_e2e_test.go
mainnet_e2e_test.go 
```